include/oneapi/tbb/detail/_concurrent_skip_list.h

```diff
@@ -700,9 +731,13 @@ class concurrent_skip_list {
     }
 
     void internal_move(concurrent_skip_list&& other) {
-        my_head_ptr.store(other.my_head_ptr.load(std::memory_order_relaxed), std::memory_order_relaxed);
+        node_ptr head = my_head_ptr.exchange(other.my_head_ptr.load(std::memory_order_relaxed), std::memory_order_relaxed);
```

Comment:
`exchange` is a relatively expensive operation on x86 (even with the relaxed semantics - it is not supported with memory operands). 
```suggestion
        node_ptr head = my_head_ptr.load(std::memory_order_relaxed);
        my_head_ptr.store(other.my_head_ptr.load(std::memory_order_relaxed), std::memory_order_relaxed);
```

Reply:
rolled back